verifier/token/cctp/attestation.go

```diff
@@ -2,42 +2,160 @@ package cctp
+) (AttestationService, error) {
+	client, err := NewHTTPClient(lggr, config)
+	if err != nil {
+		return nil, err
 	}
+	return &HTTPAttestationService{
+		lggr:               lggr,
+		client:             client,
+		ccvVerifierVersion: ccvVerifierVersion,
+		ccvAddresses:       config.ParsedVerifiers,
+	}, nil
 }
 
-func (h HTTPAttestationService) Fetch(
+// Fetch calls CCTP Attestation API using sourceChainDomain + txHash
+// It iterates through results to find the attestation for matching message
+// We use hookData and sender address to match the message against the protocol.Message.
+func (h *HTTPAttestationService) Fetch(
 	ctx context.Context,
 	txHash protocol.ByteSlice,
 	message protocol.Message,
-) (protocol.ByteSlice, error) {
-	// Call CCTPv2 using sourceChainDomain + txHash
-	// Find the attestation with the results set and return
-	// 		hookData can be found in attestation response - decodedMessage.decodedMessageBody.hookData
-	//      hookData format is the following <4 byte verifier version><32 byte msg ID>
-	return txHash, nil
+) (Attestation, error) {
+	sourceDomain, ok := sourceDomains[uint64(message.SourceChainSelector)]
+	if !ok {
+		return Attestation{}, fmt.Errorf("unsupported source chain selector: %d", message.SourceChainSelector)
+	}
+
+	response, err := h.client.GetMessages(ctx, message.SourceChainSelector, sourceDomain, txHash.String())
+	if err != nil {
+		return Attestation{}, fmt.Errorf(
+			"error fetching messages for chain selector %d and tx hash %s: %s",
+			message.SourceChainSelector, txHash, err,
+		)
+	}
+	return h.extractAttestationFromResponse(response, message)
+}
+
+func (h *HTTPAttestationService) extractAttestationFromResponse(response Messages, message protocol.Message) (Attestation, error) {
+	for _, msg := range response.Messages {
+		err := cctpMatchesMessage(h.ccvVerifierVersion, h.ccvAddresses, msg, message)
+		if err != nil {
+			h.lggr.Debugw(
+				"skipping CCTP message as it doesn't match CCIP message",
+				"message", msg,
+				"reason", err,
+			)
+			continue
+		}
+		if msg.IsAttestationComplete() {
+			return NewAttestation(h.ccvVerifierVersion, msg)
+		}
+		return Attestation{}, fmt.Errorf("attestation is not ready")
+	}
+	return Attestation{}, fmt.Errorf("no matching message found in response")
+}
+
+func cctpMatchesMessage(
+	ccvVerifierVersion protocol.ByteSlice,
+	ccvAddresses map[protocol.ChainSelector]protocol.UnknownAddress,
+	cctpMessage Message,
+	ccipMessage protocol.Message,
+) error {
+	messageID, err := ccipMessage.MessageID()
+	if err != nil {
+		return fmt.Errorf("failed to compute message ID: %w", err)
+	}
+
+	if !cctpMessage.IsV2() {
+		return fmt.Errorf("unsupported CCTP version")
+	}
+
+	ccvAddress, ok := ccvAddresses[ccipMessage.SourceChainSelector]
+	if !ok {
+		return fmt.Errorf("no CCV address configured for source chain selector: %s", ccipMessage.SourceChainSelector)
+	}
+
+	senderAddress, err := protocol.NewUnknownAddressFromHex(cctpMessage.DecodedMessage.Sender)
```

Comment:
```suggestion
	senderAddress, err := protocol.NewUnknownAddressFromHex(cctpMessage.DecodedMessage.DecodedMessageBody.MessageSender)
```
This is the field we want to validate. Top level sender would be the TokenMessenger address, a CCTP contract. The MessageSender in the message body is what actually called into the TokenMessenger.